app/services/the_tvdb_service.rb

```diff
@@ -0,0 +1,95 @@
+class TheTvdbService
+  BASE_URL = 'https://api.thetvdb.com'.freeze
+  API_KEY = ENV['THE_TVDB_API_KEY'].freeze
+
+  def initialize
+    @token = api_token
+  end
+
+  def nightly_import!
+  end
+
+  def weekly_import!
+    return if missing_data.empty?
+
+    missing_data['Anime'].each do |media_ids|
+      media_id = media_ids[0]
+
+      # check if the external_sites exist.
+      series_id = media_ids[1].try(:[], 'thetvdb/series')
+      season_id = media_ids[1].try(:[], 'thetvdb/season')
+
+      # will get the data from tvdb
+      # TODO: need to figure out how to skip if it returns nothing (404)
+      # TODO: will also need to add a check if episode_count and data.count are equal.
+      # will most likly break this out of 1 method.
+      get(build_episode_path(series_id, season_id)).try(:[], 'data')&.each do |tvdb_episode|
```

Comment:
It's possible that they're gonna force pagination on this, so you might need to do multiple requests for this.

It's possible that a sane solution might be to have one class which handles getting the TVDB IDs to do, then another class to handle using a series/season ID to get the episodes, then another to import the episodes.

That way, you keep the three pieces of logic separate: finding the Mappings, grabbing the series data, and updating the episode data